failure "empty command evaluates to undefined" with nodejs 4.8.2

Hi,

Coffeescript 1.12.5 fails when built with Nodejs 4.8.2 (the version in Debian unstable/testing).
with the following error message and stacktrace:

<pre>
bin/cake build:full
failed 1 and passed 834 tests in 23.65 seconds 

  empty command evaluates to undefined 
  AssertionError: Expected undefined to equal coffee> 
    at exports.eq (/build/coffeescript-1.12.5~dfsg/test/support/helpers.coffee:16:34)
    at /build/coffeescript-1.12.5~dfsg/test/repl.coffee:1:1
    at Function.<anonymous> (/build/coffeescript-1.12.5~dfsg/test/repl.coffee:1:1)
    at global.test (/build/coffeescript-1.12.5~dfsg/Cakefile:1:1)
    at testRepl (/build/coffeescript-1.12.5~dfsg/test/repl.coffee:1:1)
    at Object.<anonymous> (/build/coffeescript-1.12.5~dfsg/test/repl.coffee:1:1)
    at Object.<anonymous> (/build/coffeescript-1.12.5~dfsg/test/repl.coffee:1:1)
    at Module._compile (module.js:1:1)
    at Object.exports.run (/build/coffeescript-1.12.5~dfsg/lib/coffee-script/coffee-script.js:1:1)
    at runTests (/build/coffeescript-1.12.5~dfsg/Cakefile:1:1)
    at testBuiltCode (/build/coffeescript-1.12.5~dfsg/Cakefile:1:1)
    at /build/coffeescript-1.12.5~dfsg/Cakefile:1:1
    at ChildProcess.<anonymous> (/build/coffeescript-1.12.5~dfsg/Cakefile:1:1)
    at emitTwo (events.js:1:1)
    at ChildProcess.emit (events.js:1:1)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:1:1)
 
  function () {
      return fn(input, output, repl);
    }
</pre>

If instead using Nodejs 6.10.2 (in Debian experimental) the testsuite succeeds.

--- a/lib/coffee-script/repl.js
+++ b/lib/coffee-script/repl.js
@@ -173,7 +173,7 @@
         opts = {};
       }
       ref1 = process.versions.node.split('.').map(function(n) {
-        return parseInt(n);
+        return parseInt(n, 10);
       }), major = ref1[0], minor = ref1[1], build = ref1[2];
       if (major === 0 && minor < 8) {
         console.warn("Node 0.8.0+ required for CoffeeScript REPL");

--- a/src/repl.coffee
+++ b/src/repl.coffee
@@ -145,7 +145,7 @@ getCommandId = (repl, commandName) ->
 
 module.exports =
   start: (opts = {}) ->
-    [major, minor, build] = process.versions.node.split('.').map (n) -> parseInt(n)
+    [major, minor, build] = process.versions.node.split('.').map (n) -> parseInt(n, 10)
 
     if major is 0 and minor < 8
       console.warn "Node 0.8.0+ required for CoffeeScript REPL"

--- a/test/repl.coffee
+++ b/test/repl.coffee
@@ -67,6 +67,11 @@ testRepl "variables are saved", (input, output) ->
   eq "'foobar'", output.lastWrite()
 
 testRepl "empty command evaluates to undefined", (input, output) ->
+  # A regression fixed in Node 5.11.0 broke the handling of pressing enter in
+  # the Node REPL; see https://github.com/nodejs/node/pull/6090 and
+  # https://github.com/jashkenas/coffeescript/issues/4502.
+  # Just skip this test for versions of Node < 6.
+  return if parseInt(process.versions.node.split('.')[0], 10) < 6
   input.emitLine ''
   eq 'undefined', output.lastWrite()
 

